packages/server/src/server.js

```diff
@@ -158,6 +159,13 @@ export default class Server {
       }
     }
 
+    if (!this.options.dev) {
```

Comment:
It still should be executed before devMiddleware in dev mode. And the chosen renderer depends on the request modern. 
so if we move it, we can only move it to renderer instead of modern renderer and also find a way of sharing it in builder/server for devMiddleware as well

Reply:
Ah right. Having util would be also a nice idea to be shared between devMiddleware and renderer.